pkg/record/handler/save.go

```diff
@@ -182,20 +182,37 @@ func (h SaveHandler) Handle(req interface{}) (resp interface{}, err error) {
 		ErrMap: map[record.ID]skyerr.Error{},
 	}
 
-	// TODO: emit schema updated event
-	_, err = ExtendRecordSchema(h.RecordStore, h.Logger, payload.Records)
-	if err != nil {
-		h.Logger.WithError(err).Errorln("failed to migrate record schema")
-		if myerr, ok := err.(skyerr.Error); ok {
-			err = myerr
+	// Open transaction for whole operation if atomic save
+	if payload.Atomic {
+		if err = h.TxContext.BeginTx(); err != nil {
 			return
 		}
 
-		err = skyerr.NewError(skyerr.IncompatibleSchema, "failed to migrate record schema")
+		defer func() {
+			if txErr := db.EndTx(h.TxContext, err); txErr != nil {
```

Comment:
Don't resue the `err`. This defer only intended to catch error below. Separate the variable before and after.